plugins/lookup/lookup_graphql.py

```diff
@@ -0,0 +1,192 @@
+            env:
+                # in order of precedence
+                - name: NAUTOBOT_URL
+            required: True
+        validate_certs:
+            description:
+                - Whether or not to validate SSL of the Nautobot instance
+            required: False
+            default: True
+        variables:
+            description:
+                - Dictionary of keys/values to pass into the GraphQL query, see [pynautobot GraphQL documentation](https://pynautobot.readthedocs.io/en/latest/advanced/graphql.html) for more details
+            required: False
+    requirements:
+        - pynautobot
+"""
+
+EXAMPLES = """
+  # Make API Query without variables
+  - name: SET FACT OF STRING
+    set_fact:
+      query_string: |
+        query {
+          sites {
+            id
+            name
+            region {
+              name
+            }
+          }
+        }
+
+  # Make query to GraphQL Endpoint
+  - name: Obtain list of sites from Nautobot
+    set_fact:
+      query_response: "{{ query('networktocode.nautobot.lookup_graphql', query=query, url='https://nautobot.example.com', token='<redact>') }}"
+
+  # Example with variables
+  - name: SET FACTS TO SEND TO GRAPHQL ENDPOINT
+    set_fact:
+      variables:
+        site_name: den
+      query_string: |
+        query ($site_name:String!) {
+            sites (name: $site_name) {
+            id
+            name
+            region {
+                name
+            }
+            }
+        }
+
+  # Get Response with variables
+  - name: Obtain list of devices from Nautobot
+    set_fact:
+      query_response: "{{ query('networktocode.nautobot.lookup_graphql', query=query, variables=variables url='https://nautobot.example.com', token='<redact>') }}"
+"""
+
+RETURN = """
+  data:
+    description:
+      - Data result from the GraphQL endpoint
+    type: dict
+"""
+
+
+def nautobot_lookup_graphql(**kwargs):
+    """Lookup functionality, broken out to assist with testing
+
+    Returns:
+        [type]: [description]
+    """
+    # Setup API Token information, URL, and SSL verification
+    url = kwargs.get("url") or os.getenv("NAUTOBOT_URL")
+
+    # Verify URL is passed in, that it is not None
+    if url is None:
+        raise AnsibleLookupError("Missing URL of Nautobot")
+
+    token = kwargs.get("token") or os.getenv("NAUTOBOT_TOKEN")
+    ssl_verify = kwargs.get("validate_certs", True)
+
+    if not isinstance(ssl_verify, bool):
+        raise AnsibleLookupError("validate_certs must be a boolean")
+
+    nautobot_api = NautobotApiBase(token=token, url=url, ssl_verify=ssl_verify)
+    query = kwargs.get("query")
+    variables = kwargs.get("variables")
+
+    # Check that a valid query was passed in
+    if query is None:
+        raise AnsibleLookupError(
+            "Query parameter was not passed. Please verify that query is passed."
+        )
+
+    # Verify that the query is a string type
+    if not isinstance(query, str):
```

Comment:
Can it also be JSON? I haven't messed around with GraphQL much, but I thought you can pass in JSON as well? And wondering if the SDK supports it as well.

Reply:
The SDK in the current form is expecting a string. This is definitely something that needs to get tested to see if the API will take a JSON object instead of a string. I'll see if I can get testing on that early this week.

Reply:
No need to block this first iteration if it does expect a string :) We can change in the future if necessary.